src/locale/keymap-util.c

```diff
@@ -780,3 +782,141 @@ int x11_convert_to_vconsole(Context *c) {
 
         return modified;
 }
+
+static
+bool locale_encoding_is_utf8_or_unspecified(const char *locale) {
+        return !strstr(locale, ".") || endswith_no_case(locale, "UTF-8");
```

Comment:
hmm, actually thinking about this, I figure this test should probably be something like this:

```c
const char *c;
c = strchr(locale, '.');
return !c || strcaseeq(c,  ".UTF-8") || strcasestr(locale, ".UTF-8@");
```

(i.e. there must be a `.` before the `UTF-8`, and there may be either an `@` or the end of the string after it.

Reply:
I have never seen "@" used in POSIX locale strings, but some searching revealed that apparently ICU uses this for extensions - I had no idea this existed. While probably not relevant for this case, this change will make a more generally applicable function, thanks for noticing!